zebra-network/src/address_book.rs

```diff
@@ -314,6 +340,49 @@ impl AddressBook {
         meta_addr
     }
 
+    /// Returns true if `updated` needs to be applied to the recent outbound peer connection IP cache.
+    ///
+    /// Checks if there are no existing entries in the address book with this IP,
+    /// or if `updated` has a more recent update requiring the outbound connector to wait
+    /// longer before initiating handshakes with peers at this IP.
+    ///
+    /// This code only needs to check a single cache entry, rather than the entire address book,
+    /// because other code maintains these invariants:
+    /// - `last_attempt`, `last_failure`, and `last_response` times for an entry can only increase.
+    /// - these are the only fields checked by `was_recently_updated()`
```

Comment:
If we just look at last response, these docs and code need to be updated.

```suggestion
    /// - `last_response` time for an entry can only increase
    /// - this is the only field checked by `has_connection_recently_responded()`
```

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/7029/commits/788bce3b9ea89d3d5b3d535d09720b9b04c169a0.